internal/cache/lru.go

```diff
@@ -0,0 +1,119 @@
+package cache
+
+import (
+	"sync"
+)
+
+type cacheNode struct {
+	Key   chacheKey
+	Value []byte
+	Next  *cacheNode
+	Prev  *cacheNode
+}
+
+// LRUCache is a cache with the least-recently-used policy.
+type LRUCache struct {
+	hashMap  map[chacheKey]*cacheNode
+	start    *cacheNode
+	end      *cacheNode
+	count    int
+	capacity int
+	lock     sync.Mutex
```

Comment:
Is the cache being used somewhere that is accessed by multiple go routine? 

If not, I would recommend not adding a mutex at this stage.

Reply:
I think cache will be used for multiple RPC concurrently threads with certain managers

Reply:
> Is the cache being used somewhere that is accessed by multiple go routine?
> If not, I would recommend not adding a mutex at this stage.

The main purpose of the cache is to have it as a layer between the database and initially the trie, but probably all database access will use a cache, having it not thread-safe would be the same as having the db not thread-safe.